experimental/dds/tree2/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -867,29 +619,62 @@ function tryRemoveLineageEvent<T>(mark: Attach<T>, revisionToRemove: RevisionTag
 }
 
 /**
- * @returns true iff both reattaches target cells that were affected by the same detach.
- * The target cells may or may not overlap depending on detach index information.
- *
- * Only valid in the context of a rebase (i.e., both marks have the same input context).
+ * Returns a number N which encodes how the cells of the two marks are aligned.
+ * - If N is zero, then the first cell of `baseMark` is the same as the first cell of `newMark`.
+ * - If N is positive, then the first N cells of `newMark` (or all its cells if N is greater than its length)
+ * are before the first cell of `baseMark`.
+ * - If N is negative, then the first N cells of `baseMark` (or all its cells if N is greater than its length)
+ * are before the first cell of `newMark`.
  */
-function areRelatedReattaches<T>(baseMark: Reattach<T>, newMark: Reattach<T>): boolean {
-	const baseReference = baseMark.lastDetachedBy ?? baseMark.detachedBy;
-	const newReference = newMark.lastDetachedBy ?? newMark.detachedBy;
-	return baseReference === newReference;
-}
-
-/**
- * @returns true iff `baseMark` attaches nodes in cells whose contents were detached by the same change
- * that conflicts with `newMark`.
- * The target cells may or may not overlap depending on detach index information.
- */
-function isBaseAttachRelatedToConflictedDetach<T>(
-	baseMark: NewAttach<T> | Reattach<T>,
-	newMark: Detach<T> & Conflicted,
+function compareCellPositions(
 	baseRevision: RevisionTag | undefined,
-): boolean {
-	return (
-		(isActiveReattach(baseMark) && baseMark.detachedBy === newMark.conflictsWith) ||
-		(baseMark.revision ?? baseRevision) === newMark.conflictsWith
+	baseMark: EmptyInputCellMark<unknown>,
+	newMark: EmptyInputCellMark<unknown>,
+	detachOffset: number,
```

Comment:
This is populated with `reattachOffset` at the call site. The change of name is a bit confusing, but to be honest I think `reattachOffset` (and `detachOffset`) at both confusing and I keep having to look at the actual code to remind myself of what this is. I'd love it if we could find a more descriptive name. I can only come up with other not-so-good names: `gapOffsetInBase`, `detachedByBaseInGap`.

Reply:
Note that if my comment below (about the usage of `detachOffset` is correct, then this function may not need the parameter)